avocado/utils/utils_misc.py

```diff
@@ -0,0 +1,262 @@
+                    giving up (seconds)
+    :type timeout: int
+    :raises: OSError, with errno 39 in case after the timeout
+             shutil.rmtree could not successfuly complete. If any attempt
+             to rmtree fails with errno different than 39, that exception
+             will be just raised.
+    """
+    assert os.path.isdir(path), "Invalid directory to remove %s" % path
+    step = int(timeout / 10)
+    start_time = time.time()
+    success = False
+    attempts = 0
+    while int(time.time() - start_time) < timeout:
+        attempts += 1
+        try:
+            shutil.rmtree(path)
+            success = True
+            break
+        except OSError, err_info:
+            # We are only going to try if the error happened due to
+            # directory not empty (errno 39). Otherwise, raise the
+            # original exception.
+            if err_info.errno != 39:
+                raise
+            time.sleep(step)
+
+    if not success:
+        raise OSError(39,
+                      "Could not delete directory %s "
+                      "after %d s and %d attempts." %
+                      (path, timeout, attempts))
+
+
+def umount(src, mount_point, fstype, verbose=False, fstype_mtab=None):
+    """
+    Umount the src mounted in mount_point.
+
+    :src: mount source
+    :mount_point: mount point
+    :type: file system type
+    :param fstype_mtab: file system type in mtab could be different
+    :type fstype_mtab: str
+    """
+    if fstype_mtab is None:
+        fstype_mtab = fstype
+
+    if is_mounted(src, mount_point, fstype, None, verbose, fstype_mtab):
+        umount_cmd = "umount %s" % mount_point
+        try:
+            process.system(umount_cmd, verbose=verbose)
+            return True
+        except process.CmdError:
+            return False
+    else:
+        logging.debug("%s is not mounted under %s", src, mount_point)
+        return True
+
+
+def mount(src, mount_point, fstype, perm=None, verbose=False,
+          fstype_mtab=None):
+    """
+    Mount the src into mount_point of the host.
+
+    :src: mount source
+    :mount_point: mount point
+    :fstype: file system type
+    :perm: mount permission
+    :param fstype_mtab: file system type in mtab could be different
+    :type fstype_mtab: str
+    """
+    if perm is None:
+        perm = "rw"
+    if fstype_mtab is None:
+        fstype_mtab = fstype
+
+    if is_mounted(src, mount_point, fstype, perm, verbose, fstype_mtab):
+        logging.debug("%s is already mounted in %s with %s",
+                      src, mount_point, perm)
+        return True
+    mount_cmd = "mount -t %s %s %s -o %s" % (fstype, src, mount_point, perm)
+    try:
+        process.system(mount_cmd, verbose=verbose)
+    except process.CmdError:
+        return False
+    return is_mounted(src, mount_point, fstype, perm, verbose, fstype_mtab)
+
+
+def is_mounted(src, mount_point, fstype, perm=None, verbose=False,
+               fstype_mtab=None):
```

Comment:
Basically this is also a needed part. The definition looks quite good but as mentioned in `nfs` I'd add:

```
def is_mounted(src, mount_point, fstype, perm=None, verbose=False,
               fstype_mtab=None, run=process.run):
```

and use it to execute the commands. That way you could transparently support local and remote support for the `is_mounted` command.